src/HasMedia/HasMediaTrait.php

```diff
@@ -361,6 +361,10 @@ public function clearMediaCollection(string $collectionName = 'default')
      */
     public function clearMediaCollectionExcept(string $collectionName = 'default', $excludedMedia = [])
     {
+        if ($excludedMedia instanceof Media) {
+            return $this->clearMediaCollectionExceptMedia($collectionName, $excludedMedia);
```

Comment:
Seems off, doesn't this cause an infinite loop? Instead of recalling the same funtion and returnning lets just pack that single item in a collection so the rest of the function can run.

Reply:
That's a different function `clearMediaCollectionExcept ` & `clearMediaCollectionExceptMedia `